guide/guide-qip.rst

```diff
@@ -0,0 +1,264 @@
+
+.. math::
+
+   U(\Delta t)=\exp(\rm{i} \cdot \Delta t [H_d  + \sum_j u_j H_j] )
+
+To let it find the optimal pulses, we needs to give the parameters for :func:`qutip.control.pulseoptim.optimize_pulse_unitary` as keyword arguments to :meth:`qutip.qip.device.OptPulseProcessor.load_circuit`. Usually the minimal requirements are the evolution time ``evo_time`` and the number of time slices ``num_tslots`` for each gate. Other parameters can also be given in the keyword arguments. For available choices, see :func:`qutip.control.pulseoptim.optimize_pulse_unitary`. It is also possible to specify different parameters for different gates, as shown in the following example:
+
+.. code-block:: python
+
+      >>> from qutip.qip.device import OptPulseProcessor
+      >>> from qutip.qip import QubitCircuit
+      >>> from qutip import tensor, sigmaz, sigmax, sigmay
+      >>> # Same parameter for all the gates
+      ... qc = QubitCircuit(N=1)
+      >>> qc.add_gate("SNOT", 0)
+      >>>
+      >>> num_tslots = 10
+      >>> evo_time = 10
+      >>> processor = OptPulseProcessor(N=1, drift=sigmaz(), ctrls=[sigmax()])
+      >>> # num_tslots and evo_time are two keyword arguments
+      ... tlist, coeffs = processor.load_circuit(
+      ... qc, num_tslots=num_tslots, evo_time=evo_time)
+      >>>
+      >>> # Different parameters for different gates
+      ... qc = QubitCircuit(N=2)
+      >>> qc.add_gate("SNOT", 0)
+      >>> qc.add_gate("SWAP", targets=[0, 1])
+      >>> qc.add_gate('CNOT', controls=1, targets=[0])
+      >>>
+      >>> processor = OptPulseProcessor(N=2, drift=tensor([sigmaz()]*2))
+      >>> processor.add_ctrl(sigmax(), cyclic_permutation=True)
+      >>> processor.add_ctrl(sigmay(), cyclic_permutation=True)
+      >>> processor.add_ctrl(tensor([sigmay(), sigmay()]))
+      >>> setting_args = {"SNOT": {"num_tslots": 10, "evo_time": 1},
+      ...             "SWAP": {"num_tslots": 30, "evo_time": 3},
+      ...             "CNOT": {"num_tslots": 30, "evo_time": 3}}
+      >>> tlist, coeffs = processor.load_circuit(
+      ... qc, setting_args=setting_args, merge_gates=False)
+
+Noise Simulation
+================
+
+In the common way of QIP simulation, where evolution is carried out by gate matrix production, the noise is usually simulated with bit flipping and sign flipping errors. The typical approaches are either applying bit/sign flipping gate probabilistically or applying Kraus operators representing different noisy channels (e.g. amplitude damping, dephasing) after each unitary gate evolution. Those two ways are equivalent and the parameters in the Kraus operators are exactly the probability of a flipping error happens during the gate operation time.
```

Comment:
I am not sure the two approaches are equivalent from all points of view actually. They lead to similar effects though.